### PR TITLE
fix: prevent signing draft documents

### PR DESCRIPTION
**Pull request by dguyen**
## Description

Currently users can sign and complete draft documents, which will result in a completed document in an invalid state.

## Changes Made

- Prevent recipients from inserting or uninserting fields for draft documents
- Prevent recipients from completing draft documents 
- Remove ability to copy signing tokens unless document is pending

## Summary by CodeRabbit

- **New Features**
	- Enhanced document status visibility and control across various components in the application. Users can now see and interact with document statuses more dynamically in views like `DocumentPageView`, `DocumentEditPageView`, and `DocumentsDataTable`.
	- Improved document signing process with updated status checks, ensuring actions like signing, completing, and removing fields are only available under appropriate document statuses.

- **Bug Fixes**
	- Adjusted document status validation logic in server-side operations to prevent actions on incorrectly stated documents, enhancing the overall security and functionality of document processing.

### CODE REVIEW

File: packages/lib/server-only/document/complete-document-with-token.ts [hunk 1]
```diff
@@ -49,8 +49,8 @@ export const completeDocumentWithToken = async ({
 
   const document = await getDocument({ token, documentId });
 
-  if (document.status === DocumentStatus.COMPLETED) {
```
Comment (catalinpit) [inline packages/lib/server-only/document/complete-document-with-token.ts:52]: Isn't this check needed anywhere anymore?

**Review by catalinpit — COMMENTED** [review]
This "Sign" option should probably be disabled for draft docs, like "void", "download" and the others are.

CleanShot 2024-04-19 at 11 15 58

File: packages/lib/server-only/document/complete-document-with-token.ts (hunk 1 above)
Comment (dguyen) [reply packages/lib/server-only/document/complete-document-with-token.ts:52]: What do you mean? The check is still there indirectly via `status !== PENDING`

**Comment by dguyen** [comment]
> This "Sign" option should probably be disabled for draft docs, like "void", "download" and the others are.

Nice catch, thanks!

File: packages/lib/server-only/document/complete-document-with-token.ts (hunk 1 above)
Comment (catalinpit) [reply packages/lib/server-only/document/complete-document-with-token.ts:52]: Got confused with the statuses. You right.

**Review by catalinpit — APPROVED** [review]
Nice work!

**Pull request merged**